sdcm/tester.py

```diff
@@ -784,25 +784,35 @@ def run_stress_thread(self, stress_cmd, duration=None, stress_num=1, keyspace_nu
         if self.create_stats:
             self.update_stress_cmd_details(stress_cmd, prefix, stresser="cassandra-stress", aggregate=stats_aggregate_cmds)
 
-        return CassandraStressThread(loader_set=self.loaders,
-                                     stress_cmd=stress_cmd,
-                                     timeout=timeout,
-                                     output_dir=cluster.Setup.logdir(),
-                                     stress_num=stress_num,
-                                     keyspace_num=keyspace_num,
-                                     profile=profile,
-                                     node_list=self.db_cluster.nodes,
-                                     round_robin=round_robin,
-                                     keyspace_name=keyspace_name).run()
+        ret = CassandraStressThread(loader_set=self.loaders,
+                                    stress_cmd=stress_cmd,
+                                    timeout=timeout,
+                                    output_dir=cluster.Setup.logdir(),
+                                    stress_num=stress_num,
+                                    keyspace_num=keyspace_num,
+                                    profile=profile,
+                                    node_list=self.db_cluster.nodes,
+                                    round_robin=round_robin,
+                                    keyspace_name=keyspace_name).run()
+        if 'write' in stress_cmd:
+            # Configure encryption at-rest for all test tables, sleep a while to wait the workload starts and test tables are created
+            time.sleep(60)
+            self.alter_test_tables_encryption(scylla_encryption_options=self.params.get('scylla_encryption_options'))
+        return ret
 
     def run_stress_thread_bench(self, stress_cmd, duration=None, stats_aggregate_cmds=True):
 
         timeout = self.get_duration(duration)
         if self.create_stats:
             self.update_stress_cmd_details(stress_cmd, stresser="scylla-bench", aggregate=stats_aggregate_cmds)
-        return self.loaders.run_stress_thread_bench(stress_cmd, timeout,
-                                                    cluster.Setup.logdir(),
-                                                    node_list=self.db_cluster.nodes)
+        ret = self.loaders.run_stress_thread_bench(stress_cmd, timeout,
+                                                   cluster.Setup.logdir(),
+                                                   node_list=self.db_cluster.nodes)
+        if 'write' in stress_cmd:
+            # Configure encryption at-rest for all test tables, sleep a while to wait the workload starts and test tables are created
+            time.sleep(60)
```

Comment:
can this sleep go into the alter table function logic ? tests that write with no encryption should sleep here for 60sec

Reply:
The alter table function will also be used in nemesis, it doesn't need to wait the table creating.

We can check scylla_encryption_options parameter, only sleep when this parameter is assigned.